### PR TITLE
Version 8.20.0

### PR DESCRIPTION
**Pull request by SC426**
ipython 8.20.0

**Destination channel:** defaults

### Links

- [PKG-3133](https://anaconda.atlassian.net/browse/PKG-3133)
- [Upstream repository](https://github.com/ipython/ipython/tree/8.19.0)
- [Upstream changelog/diff](https://ipython.readthedocs.io/en/stable/whatsnew/index.html)

### Explanation of changes:

- Bump version to `8.20.0`
- Update build skip to `py<310`
- Add `pickleshare` to test section

[PKG-3133]: https://anaconda.atlassian.net/browse/PKG-3133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cbouss — COMMENTED** [review]
The dependency list should be updated:
https://github.com/ipython/ipython/compare/8.15.0...8.19.0#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,36 +30,32 @@ requirements:
...
     - traitlets >=5
-    - typing_extensions  # [py<310]
 
 test:
   requires:
     - pip
     {% if not migrating %}
-    #- black
+    - pytest <7.1
+    - pytest-asyncio
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - pytest-asyncio <0.22
```
https://github.com/ipython/ipython/blob/8.20.0/setup.cfg#L75C19-L75C24

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,36 +30,32 @@ requirements:
...
     - stack_data
     - traitlets >=5
-    - typing_extensions  # [py<310]
 
 test:
   requires:
     - pip
     {% if not migrating %}
-    #- black
+    - pytest <7.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - pytest
```
https://github.com/ipython/ipython/blob/8.20.0/setup.cfg#L74

Comment (SC426) [reply recipe/meta.yaml]: @skupr-anaconda Charles I couldn't get it to build locally with this configuration for `py<312`, so we opted to go with two different versions.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -30,36 +30,33 @@ requirements:
...
-    - backcall
     - colorama  # [win]
     - decorator
     - exceptiongroup  # [py<311]
     - jedi >=0.16
     - matplotlib-inline
-    - pexpect >4.3  # [unix]
-    - pickleshare
-    - prompt-toolkit >3.0.30,<3.1.0,!=3.0.37
+    - pexpect >4.3
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - pexpect >4.3  # [not win]
```

**Review by cbouss — APPROVED** [review]
LGTM

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**